### ISSUE TITLE
Bugs in ColBERT collection indexer _concatenate_and_split_sample() method

### ISSUE THREAD

**Issue by jdpsen** [primeqa/primeqa#66]
_jbross created the following on May 24:_  

 1. In https://github.ibm.com/ai-foundation/OneQA/blob/master/oneqa/ir/dense/colbert_top/colbert/indexing/collection_indexer.py#L216, the line
```
sample, sample_heldout = sample.split(int(sample.size(0) - min(.5 * sample.size(0), 50_000)), dim=0)
```

will fail with an exception in all cases where the `sample.size(0)` is < 50_000 and is an odd number:

- The torch.tensor.split(int) method will split a tensor into equal sized sub tensors along the given dimension, where the last sub tensor might be smaller.
- In case that the sample size is odd and < 50_000, the above line of code will always result in 3 splits (last split has length 1), and an exception will be raised as code is trying to assign 3 values to just 2 variables (sample, sample_heldout = ...).

2. The comment in https://github.ibm.com/ai-foundation/OneQA/blob/master/oneqa/ir/dense/colbert_top/colbert/indexing/collection_indexer.py#L211 says
 "# Shuffle and split out a 5% "heldout" sub-sample [up to 50k elements]", but the code actually splits out 50% (not 5%) of the data!

Suggested fix for both issues:

```
heldout_fraction = 0.05 # 5%
heldout_size = int(min(heldout_fraction * sample.size(0), 50_000))
sample, sample_heldout = sample.split([sample.size(0) - heldout_size, heldout_size], dim=0)
```
 

_Original issue: https://github.ibm.com/ai-foundation/PrimeQA/issues/214_

**Comment by jdpsen** [comment]
_On May 26, franzm commented:_
@jbross good catch, thanks, I've also shared it with the Stanford people.

**Comment by jdpsen** [comment]
@franzmpub : please update the status and the pipeline label for this issue

**Comment by franzmpub** [comment]
Fixed in PR 270
